### PR TITLE
#81 Get CH costs

### PR DESCRIPTION
**Pull request by catherineholland1**
conversion of script: CH costs lookup

#81

### CODE REVIEW

File: Production_scripts/get_ch_costs.R [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  paste0(get_slf_dir(), "/Costs/CH_Costs.xlsx")
+)
+
+
+ch <-
+  ch %>%
+  # rename
+  rename(source_of_funding = "Source of Funding") %>%
+  # select only the funding totals
+  filter(source_of_funding == "All Funding With Nursing Care" | source_of_funding == "All Funding Without Nursing Care")
```
Comment (Moohan) [inline Production_scripts/get_ch_costs.R:44]:
```suggestion
  filter(source_of_funding %in% c("All Funding With Nursing Care", "All Funding Without Nursing Care"))
```

File: Production_scripts/get_ch_costs.R [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+haven::write_sav(outfile,
+  paste0(get_slf_dir(), "/Costs/Cost_CH_Lookup.sav"),
+  compress = TRUE
+)
+
+# .rds file
+readr::write_rds(outfile,
+  paste0(get_slf_dir(), "/Costs/Cost_CH_Lookup.sav"),
+  compress = "gz"
+)
```
Comment (Moohan) [inline Production_scripts/get_ch_costs.R:179]: Can use `get_ch_costs_path()` here

**Review by Moohan — CHANGES_REQUESTED** [review]
Sorry for the short comments! Just trying to have a quick look at all the PRs before I finish.

**Comment by catherineholland1** [comment]
ready for your review @Jennit07

File: Production_scripts/Costs-01-care_home.R [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  select(
+    year,
+    nursing_care_provision,
+    cost_per_day
+  )
+
+
+# .zsav
+haven::write_sav(outfile,
+  get_ch_costs_path(),
```
Comment (Jennit07) [inline Production_scripts/Costs-01-care_home.R:118]: do we need to specify update = latest_update() here?
```suggestion
  get_ch_costs_path(update = latest_update()),
```

File: Production_scripts/Costs-01-care_home.R [hunk 4]
```diff
@@ -0,0 +1,171 @@
+#####################################################
```
Comment (Jennit07) [inline Production_scripts/Costs-01-care_home.R:1]:
```suggestion
#####################################################
# Costs - Care Home
# Author: Catherine Holland
# Date: February 2022
# Written on RStudio Server
# Version of R - 3.6.1
# Input - COSLA Values
# Description - Lookup for costs for Care Homes
#
# Get COSLA Value tables
# Estimated Average Gross Weekly Charge for Long Stay Residents in Care Homes for Older People in Scotland.
# https://publichealthscotland.scot/publications/care-home-census-for-adults-in-scotland/
# Check and add any new years to 'CH_Costs.xlsx'
#
#####################################################

# Load packages
library(dplyr)
library(createslf)

# Read in data---------------------------------------

## Make a copy of the existing file
fs::file_copy(get_ch_costs_path(), get_ch_costs_path(update = previous_update()))

## Read excel data
ch <- readxl::read_xlsx(
  paste0(get_slf_dir(), "/Costs/CH_Costs.xlsx")
)


# Data cleaning ---------------------------------------
ch <-
  ch %>%
  # rename
  rename(source_of_funding = "Source of Funding") %>%
  # select only the funding totals
  filter(source_of_funding %in% c("All Funding With Nursing Care", "All Funding Without Nursing Care")) %>%
  # restructure
  pivot_longer(
    !contains("source"),
    names_to = "calendar_year",
    values_to = "cost_per_week"
  ) %>%
  # create year as FY = YYYY from CCYY
  mutate(financial_year = convert_year_to_fyyear(calendar_year)) %>%
  # create flag - nursing care provision ##
  mutate(nursing_care_provision = if_else(source_of_funding == "All Funding With Nursing Care", 1, 0)) %>%
  # cost per day ##
  mutate(cost_per_day = cost_per_week / 7) %>%
  # unknown nursing variable ##
  mutate(unknown_nursing = "Unknown Source of Funding") %>%
  # restructure
  pivot_longer(
    cols = any_of(c("source_of_funding", "unknown_nursing")),
    values_to = "source_of_funding",
    names_to = NULL
  ) %>%
  # replace nursing care provision with NA
  mutate(nursing_care_provision = na_if(nursing_care_provision, source_of_funding == "Unknown Source of Funding"))

# aggregate
outfile <-
  ch %>%
  group_by(financial_year, nursing_care_provision) %>%
  mutate(cost_per_day = mean(cost_per_day)) %>%
  select(financial_year, nursing_care_provision, cost_per_day)


## add in years by copying the most recent year ##
latest_year <- 2122

## increase by 1% for every year after the latest ##
outfile <-
  map_df(1:5, ~
  outfile %>%
    group_by(financial_year, nursing_care_provision) %>%
    arrange(financial_year, nursing_care_provision) %>%
    mutate(
      cost_per_day = cost_per_day * (1.01)^.x,
      financial_year = convert_year_to_fyyear(as.numeric(convert_fyyear_to_year(financial_year)) + .x)
    )) %>%
  arrange(financial_year, nursing_care_provision) %>%
  rename(year = "financial_year")


# Join data together  -----------------------------------------------------

# match files - to make sure costs haven't changed radically
lookup <- haven::read_sav(
  find_latest_file(get_slf_dir(), regexp = "Cost_CH_Lookup_pre.+?\\.sav")
) %>%
  rename(
    cost_old = "cost_per_day",
    year = "Year"
  ) %>%
  arrange(year, nursing_care_provision) %>%
  # match to new costs
  full_join(lookup, by = c("year", "nursing_care_provision")) %>%
  # compute difference
  mutate(pct_diff = (cost_per_day - cost_old) / cost_old * 100) %>%
  # count
  count(pct_diff, year, nursing_care_provision) %>%
  spread(year, n)


## save outfile ---------------------------------------
outfile <-
  data %>%
  select(
    year,
    nursing_care_provision,
    cost_per_day
  )

# .zsav
haven::write_sav(outfile,
  get_ch_costs_path(),
  compress = TRUE
)

# .rds file
readr::write_rds(outfile,
  get_ch_costs_path(),
  compress = "gz"
)

## End of Script ---------------------------------------
```
I've split the script into sections and used the pipe to make the code flow without having to specify data %>% data etc. This is just in line with other scripts. Will need testing to make sure the code isnt breaking after the changes :)

**Review by Jennit07 — CHANGES_REQUESTED** [review]
Code looks good. Usual suggestions for restructuring the script into sections. Can you please test to make sure you are happy with what the code is doing/nothing breaking. Thanks :)

**Review by catherineholland1 — COMMENTED** [review]
changes complete - just wee thing to do with file name. 

Had to remove one line from pipe as was then changing data file to not the format needed to save, I think? Code all working as expecting

**Comment by Jennit07** [comment]
Costs paths have now been updated. Will need to use get_ch_costs_path(update = latest_update()) to save outfile

**Review by Moohan — APPROVED** [review]

**Comment by Moohan** [comment]
@catherineholland1 I just made quite a few changes, particularly to fix the uplift code which wasn't quite right. Could you have a look at them and if you're happy that it makes sense and works then merge, please? If you want to make any other changes or notice that I broke something then please fix/request another review!

**Review by catherineholland1 — COMMENTED** [review]
all looks good thanks James and if does what it is meant to then happy to merge!

**Review by catherineholland1 — COMMENTED** [review]
ready to merge

**Pull request merged**